src/main/java/org/jabref/gui/externalfiles/ImportHandler.java

```diff
@@ -118,8 +125,10 @@ public List<ImportFilesResultItemViewModel> call() {
                     }
 
                     UiTaskExecutor.runInJavaFXThread(() -> {
-                        updateMessage(Localization.lang("Processing file %0", file.getFileName()));
-                        updateProgress(counter, files.size() - 1d);
+                        setTitle(Localization.lang("Processing %0 file(s) for adding to %1.", files.size(), bibDatabaseContext.getDatabasePath().map(path -> path.getFileName().toString()).orElseGet(() -> "untitled")));
+                        updateMessage(Localization.lang("Processing file %0 | %1 of %2 file(s) processed.", file.getFileName(), counter, files.size()));
```

Comment:
When the file name is too long, users have to scroll horizontaly in the task viewer.

It would be nice if you could introduce a function to reduce long strings into fixed size strings by cutting the end and replacing it with "...".

Other JabRef features could benefit from this function :)

Reply:
Be awre that the filename is the most important part and comes last. Should not be replaced by ...

Otherwise StringUtil.limitStringLength is the function to use.

Reply:
Thank you for the feedback. I will work on an implementation that prominently displays the filename because it is the most important part and modifies the message so that users do not have to scroll horizontally.